modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -31,6 +31,110 @@
 )
 
 
+def lazy_metadata_decorator(
+    apply_axis=None, inherit=False, axis_arg=-1, transpose=False
+):
+    """
+    Lazily propagate metadata for the ``PandasDataframe``.
+
+    Parameters
+    ----------
+    apply_axis : str, default: None
+        The axes on which to apply the index object to the `self._partitions` lazily.
+    inherit : bool, default: False
```

Comment:
I think with `apply_axis=None`, there might be some confusion because in `synchronize_labels` and other use cases None implies both row and column axes. With regards to your point on `apply_axis="same"` and `axis=0` and other such examples, that is more axis/operation specific. What are your thoughts on `inherit_both_axes`?

Reply:
I think it's okay for the `None` argument to mean different things in `lazy_metadata_decorator`, `synchronize_labels`, and `_propagate_index_objs`. Keeping the options to a single argument would also preclude the meaningless subset of arguments `apply_axis="rows"` and `inherit_both_axes=True`. I won't insist on this change, but if you want the bool, I prefer `inherit_both_axes` to `inherit`.